test/index_test.js

```diff
@@ -190,12 +190,98 @@ tape.test('v5 compile', function(t) {
-        }, /language code foo not loaded/
-    );
+        }, /language code foo not loaded/);
 
         assert.end();
     });
 
+    t.test('en-US fallback to en', function(assert) {
+        var v5Compiler = compiler('v5');
+        var language = v5Compiler.getBestMatchingLanguage('en-us');
+
+        assert.equal(v5Compiler.compile(language, {
+            maneuver: {
+                type: 'turn',
+                modifier: 'left'
+            },
+            name: 'Way Name'
+        }), 'Turn left onto Way Name');
+
+        assert.end();
+    });
+
+    t.test('zh-CN fallback to zh-Hans', function(assert) {
+        var v5Compiler = compiler('v5');
+        var language = v5Compiler.getBestMatchingLanguage('zh-CN');
+
+        assert.equal(v5Compiler.compile(language, {
+            maneuver: {
+                type: 'turn',
+                modifier: 'left'
+            },
+            name: 'Way Name'
+        }), '左转，上Way Name');
+
+        assert.end();
+    });
+
+    t.test('zh-Hant fallback to zh-Hanz', function(assert) {
+        var v5Compiler = compiler('v5');
+        var language = v5Compiler.getBestMatchingLanguage('zh-Hant');
+
+        assert.equal(v5Compiler.compile(language, {
+            maneuver: {
+                type: 'turn',
+                modifier: 'left'
+            },
+            name: 'Way Name'
+        }), '左转，上Way Name');
+
+        assert.end();
+    });
+
+    t.test('zh-Hant-TW fallback to zh-Hant', function(assert) {
+        var v5Compiler = compiler('v5');
+        var language = v5Compiler.getBestMatchingLanguage('zh-Hant-TW');
+
+        assert.equal(v5Compiler.compile(language, {
+            maneuver: {
+                type: 'turn',
+                modifier: 'left'
+            },
+            name: 'Way Name'
+        }), '左转，上Way Name');
+
+        assert.end();
+    });
+
+    t.test('es-MX fallback to es', function(assert) {
+        var v5Compiler = compiler('v5');
+        var language = v5Compiler.getBestMatchingLanguage('es-MX');
+
+        assert.equal(v5Compiler.compile(language, {
+            maneuver: {
+                type: 'turn',
+                modifier: 'straight'
+            },
+            name: 'Way Name'
+        }), 'Ve recto en Way Name');
+
+        assert.end();
+    });
+
+    t.test('getBestMatchingLanguage', function(t) {
+        t.assert(compiler('v5').getBestMatchingLanguage('foo'), 'en');
+        t.assert(compiler('v5').getBestMatchingLanguage('en-US'), 'en');
+        t.assert(compiler('v5').getBestMatchingLanguage('zh-CN'), 'zh-Hans');
+        t.assert(compiler('v5').getBestMatchingLanguage('zh-Hant'), 'zh-Hans');
+        t.assert(compiler('v5').getBestMatchingLanguage('zh-Hant-TW'), 'zh-Hans');
+        t.assert(compiler('v5').getBestMatchingLanguage('zh'), 'zh-Hans');
+        t.assert(compiler('v5').getBestMatchingLanguage('es-MX'), 'es');
+        t.assert(compiler('v5').getBestMatchingLanguage('es-es'), 'es-es');
+        t.end();
+    });
```

Comment:
Let’s add some tests of `divideLanguageCodes()` or whatever we call it once we move it to languages.js.

Reply:
Added here: https://github.com/Project-OSRM/osrm-text-instructions/pull/168/files#diff-979945c097652b2859f6887cda2f7178R288